src/DocBlock/Tags/TemplateExtends.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * This file is part of phpDocumentor.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @link http://phpdoc.org
+ */
+
+namespace phpDocumentor\Reflection\DocBlock\Tags;
+
+use phpDocumentor\Reflection\Type;
+use phpDocumentor\Reflection\DocBlock\Description;
+
+/**
+ * Reflection class for a {@}template-extends tag in a Docblock.
+ */
+final class TemplateExtends extends Extends_ 
```

Comment:
Is Template extends a decorator on extends? And could you replace template-extends with extends? Because that's what it means when you extend a tag this way. I'm not sure that's what you intended to do.

In this project we chose often to duplicate code because tags are not the same and cannot be interchanged. If we want to share code between tags we chose to create a base class. 

I'm not aware of what template-extends does. And I couldn't find it in the phpstan documentation. So it could be that you are fully correct here.

Reply:
Well read [this](https://psalm.dev/docs/annotating_code/templated_annotations)